src/nonce.ts

```diff
@@ -22,7 +23,7 @@ export class Nonce {
 
   constructor (n = MIN_NONCE) {
     this.n = n
-    this.bytes = new Uint8Array(12)
+    this.bytes = allocUnsafe(12)
```

Comment:
this might be dangerous as it could already contain some data.

Reply:
we immediately set value below so I think it's safe for this place but you're right, will need to review all other places

Reply:
notice that in other dependencies of js-libp2p we do both `alloc()` and `allocUnsafe()` so we may want to do the same thing https://github.com/achingbrain/uint8arrays/blob/56329d16d6ca575c9638f3abb9601b8a034783b8/src/alloc.js#L8